tests/main/parallel-install-classic/task.yaml

```
@@ -0,0 +1,88 @@
+summary: Checks parallel installation of classic snaps
+
+# core does not support classic confinement
+systems: [-ubuntu-core-*]
+
+prepare: |
+    # ensure we have no snap user data directories yet
+    rm -rf /home/test/snap
+    rm -rf /root/snap
+    rm -rf /var/snap/test-snapd-classic-confinement
+    rm -rf /var/snap/test-snapd-classic-confinement_foo
+
+    #shellcheck source=tests/lib/snaps.sh
+    . "$TESTSLIB"/snaps.sh
+    #shellcheck source=tests/lib/dirs.sh
+    . "$TESTSLIB"/dirs.sh
+    snap pack "$TESTSLIB/snaps/test-snapd-classic-confinement/"
+
+    case "$SPREAD_SYSTEM" in
+        fedora-*|arch-*|centos-*)
+            # although classic snaps do not work out of the box on fedora,
+            # we still want to verify if the basics do work if the user
+            # symlinks /snap to $SNAP_MOUNT_DIR themselves
+            ln -sf "$SNAP_MOUNT_DIR" /snap
+            ;;
+    esac
+
+    snap set system experimental.parallel-instances=true
+
+restore:
+    snap set system experimental.parallel-instances=null
+
+    case "$SPREAD_SYSTEM" in
+        fedora-*|arch-*|centos-*)
+            rm -f /snap
+            ;;
+    esac
+
+execute: |
+    #shellcheck source=tests/lib/snaps.sh
+    . "$TESTSLIB"/snaps.sh
+
+    install_local test-snapd-classic-confinement --classic
+    install_local_as test-snapd-classic-confinement test-snapd-classic-confinement_foo --classic
+
+    su -l -c '! test -d ~/snap/test-snapd-classic-confinement' test
```

You can use a slightly less confusing way to run shell as a test user, where the "test" user doesn't need to be annoyingly at the end of the command:

```suggestion
    su test -c '! test -d ~/snap/test-snapd-classic-confinement'
```

You can apply this pattern to the remainder of the test.

---

Applied with a tweak.